.github/workflows/build-depends.yml

```diff
@@ -14,80 +14,107 @@ on:
     outputs:
       key:
         description: "Key needed for restoring depends cache"
-        value: ${{ jobs.build-depends.outputs.key }}
+        value: ${{ jobs.check-cache.outputs.cache-key }}
+      host:
+        description: "Host triplet for this build target"
+        value: ${{ jobs.check-cache.outputs.host }}
+      dep-opts:
+        description: "DEP_OPTS used to build depends"
+        value: ${{ jobs.check-cache.outputs.dep-opts }}
 
 jobs:
-  build-depends:
-    name: Build depends
-    runs-on: ubuntu-24.04
+  check-cache:
+    name: Check cache
+    runs-on: ubuntu-latest
     outputs:
-      key:  ${{ steps.restore.outputs.cache-primary-key }}
-    container:
-      image: ${{ inputs.container-path }}
-      options: --user root
+      cache-hit: ${{ steps.cache-check.outputs.cache-hit }}
+      cache-key: ${{ steps.setup.outputs.cache-key }}
+      host: ${{ steps.setup.outputs.HOST }}
+      dep-opts: ${{ steps.setup.outputs.DEP_OPTS }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
           ref: ${{ github.event.pull_request.head.sha }}
+          sparse-checkout: |
+            ci/dash
+            ci/test
+            depends/packages
+            depends/hosts
+            contrib/containers/ci/ci.Dockerfile
+            contrib/containers/ci/ci-slim.Dockerfile
 
-      - name: Initial setup
+      - name: Compute cache key
         id: setup
         run: |
           BUILD_TARGET="${{ inputs.build-target }}"
           source ./ci/dash/matrix.sh
           echo "DEP_OPTS=${DEP_OPTS}" >> "${GITHUB_OUTPUT}"
           echo "HOST=${HOST}" >> "${GITHUB_OUTPUT}"
           DEP_HASH="$(echo -n "${BUILD_TARGET}" "${DEP_OPTS}" "${HOST}" | sha256sum | head -c 64)"
-          echo "\"${BUILD_TARGET}\" has HOST=\"${HOST}\" and DEP_OPTS=\"${DEP_OPTS}\" with hash \"${DEP_HASH}\""
           echo "DEP_HASH=${DEP_HASH}" >> "${GITHUB_OUTPUT}"
-
+          DOCKERFILE_HASH="${{ hashFiles('contrib/containers/ci/ci.Dockerfile', 'contrib/containers/ci/ci-slim.Dockerfile') }}"
+          PACKAGES_HASH="${{ hashFiles('depends/packages/*') }}"
```

Comment:
should add `depends/Makefile` too. It is changed rarely, but it happens couple times each year

Reply:
should be this file `.github/workflows/build-depends.yml` also be part of hash?